Userland/Libraries/LibVirtGPU/CommandBufferBuilder.cpp

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright (c) 2022, Sahan Fernando <sahan.h.fernando@gmail.com>
+ * Copyright (c) 2022, Stephan Unverwerth <s.unverwerth@serenityos.org>
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/StringView.h>
+#include <Kernel/API/VirGL.h>
+#include <sys/ioctl_numbers.h>
+
+#include <LibVirtGPU/CommandBufferBuilder.h>
+#include <LibVirtGPU/Commands.h>
+#include <LibVirtGPU/VirGLProtocol.h>
+
+namespace VirtGPU {
+
+static u32 encode_command(u16 length, Protocol::ObjectType object_type, Protocol::VirGLCommand command)
+{
```

Comment:
I'm not sure if mid is always an object type, but I guess this should be fine for now.